tests/sympc/tensor/mpc_tensor_test.py

```diff
@@ -200,3 +207,43 @@ def test_mpc_print(get_clients) -> None:
     expected = f"{expected} <VirtualMachineClient: P_1 Client> -> ShareTensorPointer"
 
     assert expected == x.__str__()
+    assert x.__str__() == x.__str__()
+
+
+def test_generate_shares(get_clients) -> None:
+
+    precision = 12
+    base = 4
+
+    x_secret = torch.Tensor([5.0])
+
+    # test with default values
+    x_share = ShareTensor(data=x_secret)
+
+    shares_from_share_tensor = MPCTensor.generate_shares(x_share, 2)
+    shares_from_secret = MPCTensor.generate_shares(x_secret, 2)
+
+    assert sum(shares_from_share_tensor).tensor == sum(shares_from_secret).tensor
```

Comment:
I think we can simply do the sum without checking the tensor?

Reply:
In this case we have to check only the tensor because the session is different and in `__eq__` method we check both.

I have  other test `test_generate_shares_session`  where I provide the session, and in this case I follow your recommendation.